apps/api/src/server/migration/migrators/nsip-project-update-migrator.js

```diff
@@ -0,0 +1,184 @@
+import { databaseConnector } from '#utils/database-connector.js';
+import sanitizeHtml from 'sanitize-html';
+import { allowedTags } from '../../applications/application/project-updates/project-updates.validators.js';
+import { buildProjectUpdatePayload } from '../../applications/application/project-updates/project-updates.mapper.js';
+import { eventClient } from '#infrastructure/event-client.js';
+import { NSIP_PROJECT_UPDATE } from '#infrastructure/topics.js';
+/**
+ * @typedef {import('../../../message-schemas/events/nsip-project-update.d.ts').NSIPProjectUpdate} NSIPProjectUpdate
+ */
+
+/**
+ * @typedef {Object} NSIPProjectUpdateCaseData
+ * @property {string} caseReference
+ * @property {string} caseName
+ * @property {string} caseDescription
+ * @property {string} caseStage
+ *
+ * @typedef {NSIPProjectUpdate & NSIPProjectUpdateCaseData} NSIPProjectUpdateMigrateModel
+ */
+
+/**
+ * @typedef {Object} UpdateToBroadcast
+ * @property {string} caseReference
+ * @property {import('@prisma/client').ProjectUpdate} updatedEntity
+ */
+
+/**
+ * Handle an HTTP trigger/request to run the migration
+ *
+ * @param {NSIPProjectUpdateMigrateModel[]} projectUpdates
+ */
+export const migrateNsipProjectUpdates = async (projectUpdates) => {
+	console.info(`Migrating ${projectUpdates.length} project updates`);
+
+	/** @type {UpdateToBroadcast[]} */
+	const updatesToBroadcast = [];
+
+	for (const model of projectUpdates) {
```

Comment:
Do we want to try/catch the logic in the for, so we can still process any that succeed?

Reply:
So the design is currently "if any fail then they all fail", and we propagate an error back to the front-end. This is just because it makes the control flow easier, since the end-user will have to fix something and since the operation is idempotent they can try everything again.

If any fail to insert, I don't think we should broadcast any events - because the front-end would wind up with a partial list of banners, so I just thought abandoning ship entirely was easier.

What are your thoughts?